pkg/api/steve/proxy/path.go

```diff
@@ -0,0 +1,70 @@
+package proxy
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var errBadResource = errors.New("unidentified resource")
+var errBadGVR = errors.New("invalid GVR")
+
+// gvrFromPath tries to parse a URL path and extract GVR information from it.
+// Version must be present.
+// Group may be empty for core discovery client paths.
+// Resource may be empty for non-core discovery client paths.
+// This won't handle paths where "namespaces" designates a custom Namespace kind. So this function is not generally
+// applicable for any paths including any resources.
+func gvrFromPath(path string) (*schema.GroupVersionResource, error) {
+	parts := pathParts(path)
+	// Allow paths for core resource discovery, such as "/api/v1". And disallow non-core short paths, such as "/apis/management.cattle.io".
+	if len(parts) < 2 || (len(parts) < 3 && parts[0] != "api") {
+		return nil, fmt.Errorf("%w for path %s", errBadGVR, path)
+	}
+	var gvr schema.GroupVersionResource
+	var resourcePath []string
+	if parts[0] == "api" {
+		gvr.Version = parts[1]
+		resourcePath = parts[2:]
+	} else {
+		gvr.Group = parts[1]
+		gvr.Version = parts[2]
```

Comment:
I might be missing something, but can't this panic in the case of a path exactly 2 entries long? We won't exist early, and if the first part isn't api, we would attempt to get the version, which would panic.

Reply:
No, one of the top tests verifies that `"/apis/management.cattle.io"` returns an error.

Reply:
Your example has a length of 3, not of two.

```
	split := strings.Split("/apis/management.cattle.io", "/")
	fmt.Printf("%d", len(split))
```
That prints `3`.

Reply:
Right, but I don't use a raw call to `strings.Split`. I filter out those empty strings, so the length is 2 after I get the parts from `pathParts`.

Reply:
Thanks, that makes sense. I think the first if statement is protecting this.

I'll resolve, though I don't like how precarious this balance is.